### PR TITLE
Upgrade Eslint to v9

### LINKED ISSUE #689: Update ESLint and various plugins to v9 ecosystem
We need to update our ESLint setup to v9, which will mean rewriting the config to a flat config, finding out of our plugins still work, etc.

### PR DESCRIPTION
**Pull request by menghif**
Fixes #689

This PR upgrades Eslint to v9 and moves to a flat config.

All plugins are now up to date and compatible with the new Eslint flat config. The exception was `eslint-plugin-react-hooks` where I had to use the latest rc version (https://github.com/facebook/react/issues/28313#issuecomment-2363692765).

I started the upgrade process with the [configuration migrator](https://eslint.org/docs/latest/use/configure/migration-guide#migrate-your-config-file) and then went through each plugin and used the recommended configuration setup found in the docs (while making sure all the original chatcraft custom configurations are kept).

### CODE REVIEW

File: src/components/CodeHeader.tsx [hunk 1]
```diff
@@ -84,6 +84,7 @@ function CodeHeader({
     try {
       // val.town returns an empty body when code doesn't return a value, which breaks res.json()
       result = await res.json();
+      // eslint-disable-next-line @typescript-eslint/no-unused-vars
     } catch (err) {
```
Comment (humphd) [inline src/components/CodeHeader.tsx:88]: You could change this to `} catch {` if we aren't using it vs. disabling the rule

**Review by humphd — CHANGES_REQUESTED** [review]
Excellent.  Those `catch(err)` issues could be converted to not require the disable.

**Comment by humphd** [comment]
Cloudflare build fails like so:

```

16:56:24.488 | > chatcraft.org@0.0.0 build /opt/buildhome/repo
-- | --
16:56:24.488 | > tsc && vite build
16:56:24.488 |  
16:56:32.623 | src/components/Sidebar/SidebarDesktop.tsx(1,15): error TS2305: Module '"@chakra-ui/react"' has no exported member 'keyframes'.
16:56:32.658 | ELIFECYCLE  Command failed with exit code 2.
16:56:32.690 | Failed: Error while executing user command. Exited with error code: 2
16:56:32.700 | Failed: build command exited with code: 1
16:56:33.704 | Failed: error occurred while running build command
```

**Comment by menghif** [comment]
> Cloudflare build fails like so:
> 
> ```
> 
> 16:56:24.488 | > chatcraft.org@0.0.0 build /opt/buildhome/repo
> -- | --
> 16:56:24.488 | > tsc && vite build
> 16:56:24.488 |  
> 16:56:32.623 | src/components/Sidebar/SidebarDesktop.tsx(1,15): error TS2305: Module '"@chakra-ui/react"' has no exported member 'keyframes'.
> 16:56:32.658 | ELIFECYCLE  Command failed with exit code 2.
> 16:56:32.690 | Failed: Error while executing user command. Exited with error code: 2
> 16:56:32.700 | Failed: build command exited with code: 1
> 16:56:33.704 | Failed: error occurred while running build command
> ```

This is because I upgraded `@chakra-ui/react` without noticing. I reverted it now!

**Review by humphd — APPROVED** [review]
Excellent

**Pull request merged**